src/main/controllers/RespondentAddressController.ts

```diff
@@ -109,13 +111,13 @@ export default class RespondentAddressController {
     const { saveForLater } = req.body;
     if (saveForLater) {
       handleSessionErrors(req, res, this.form, PageUrls.CLAIM_SAVED);
-      handleUpdateDraftCase(req, this.logger);
+      handleUpdateDraftCase(req, logger);
       return res.redirect(PageUrls.CLAIM_SAVED);
     } else {
       const nextPage = req.session.userCase.respondents.length > 1 ? PageUrls.ACAS_CERT_NUM : PageUrls.WORK_ADDRESS;
       const redirectUrl = getRespondentRedirectUrl(req.params.respondentNumber, nextPage);
       handleSessionErrors(req, res, this.form, redirectUrl);
-      handleUpdateDraftCase(req, this.logger);
+      handleUpdateDraftCase(req, logger);
```

Comment:
Same here, was this a previous pattern we used?

Reply:
I started looking at this and got into a bit of a rabbit hole @marcjudge:
When we handleSessionErrors on a case, if we only save for later then we redirect to claim-saved page even if there are errors (as long as some of them are required) but the RespondentAddressController will redirect to claim-saved page either way (even if all required fields have been inputted). 
Is this intended? It feels like it was missed when updating other controllers - and i'm sure there are others like it.

Reply:
I think we need to revisit the 'saveForLater' button, as I'm not sure the behaviour is consistent from page to page. It's also not clear what the behaviour should be. From requirements gathering a while back the business requested no validation rules be applied when you select 'saveForLater'. They didn't want the user to get stuck if they couldn't answer a question.
It looks like 'handleSessionErrors' is being called on multiple pages when you select 'saveForLater' which is not correct as it will reload the same page if it finds any errors. There should probably be a different error checking method for the 'saveForLater' button which returns a boolean that determines if handleUpdateDraftCase is called.

Reply:
Created this placeholder - https://tools.hmcts.net/jira/browse/RET-2747